### PR TITLE
Manual DS4 mode: Allow Back/Select to trigger Touchpad click

### PR DESCRIPTION
**Pull request by ekchang**
## Description
When forcing manual DS4 emulation, there isn't a natural way to trigger Touchpad click for standard XInput. Current impl supports passing MISC_BUTTON (Xbox Series X Share button), but Steam Input doesn't expose this mapping, and most other XInput controllers on client don't have this button available.

Allow additional accessibility for clients wishing to emulate DS4 by having Back/Select also trigger Touchpad click. In most games supporting native DS4 input, Back/Share does not do anything anyway and typically should translate to a Touchpad click.

My specific usecase here is using Sunshine via Steam Deck Moonlight client. A DualShock 4 or DualSense controller configured via Steam Input cannot map the Touchpad click to Xbox 'Share' and the only way to pass through Touchpad clicks through to Sunshine is by disabling Steam Input altogether. 

### Screenshot
Screenshot 2023-10-21 194002

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: src/platform/windows/input.cpp [hunk 1]
```diff
@@ -1333,6 +1333,11 @@ namespace platf {
 
     // Allow either PS4/PS5 clickpad button or Xbox Series X share button to activate DS4 clickpad
     if (gamepad_state.buttonFlags & (TOUCHPAD_BUTTON | MISC_BUTTON)) buttons |= DS4_SPECIAL_BUTTON_TOUCHPAD;
+    
+    // Manual DS4 emulation: check if BACK button should also trigger DS4 touchpad click
+    if ((config::input.gamepad == "ps4"sv || config::input.gamepad == "ds4"sv)
```
Comment (ReenigneArcher) [inline src/platform/windows/input.cpp:1338]: Unless something changed recently, I don't think `ps4` is a valid option.

File: src_assets/common/assets/web/config.html [hunk 2]
```diff
@@ -105,6 +105,29 @@ <h1 class="my-4">Configuration</h1>
         </select>
         <div class="form-text">Choose which type of gamepad to emulate on the host</div>
       </div>
+      <div class="accordion" v-if="config.gamepad === 'ds4'">
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/config.html:108]: Does this auto update when the gamepad option is changed? I had an issue recently where I did something similar and the ui didn't update upon change.

File: src_assets/common/assets/web/config.html [hunk 3]
```diff
@@ -1140,6 +1163,7 @@ <h2 class="accordion-header">
     "encoder": "",
     "fps": "[10,30,60,90,120]",
     "gamepad": "auto",
+    "ds4_back_as_touchpad": "enabled",
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/config.html:1166]: Alphabetical please.

**Review by ReenigneArcher — CHANGES_REQUESTED** [review]
Thank you for the PR!

In addition to the below comments. I would also request the docs are updated. All config options are in the "advanced usage" section.

Also, perhaps this option should be named "ds4_back_as_touchpad_click" or something along those lines?

File: src/platform/windows/input.cpp (hunk 1 above)
Comment (ekchang) [reply src/platform/windows/input.cpp:1338]: Yeah I wasn't sure if that was still valid so I kept it consistent with https://github.com/ekchang/Sunshine/blob/5b052d7c2c059aa26fdf79cc87885af2f461f544/src/platform/windows/input.cpp#L1177-L1179

But I'll go ahead and update if we're confident ds4 is the only supported config here

File: src_assets/common/assets/web/config.html (hunk 2 above)
Comment (ekchang) [reply src_assets/common/assets/web/config.html:108]: It does! I can confirm that switching to auto and x360 will collapse this accordion from being visible in realtime.

File: src/platform/windows/input.cpp (hunk 1 above)
Comment (ReenigneArcher) [reply src/platform/windows/input.cpp:1338]: Thanks for the background... I think we should probably remove "ps4" from there as well. Would you take care of it in this PR?

File: src_assets/common/assets/web/config.html (hunk 2 above)
Comment (ReenigneArcher) [reply src_assets/common/assets/web/config.html:108]: Great, thanks for confirming!

**Comment by ekchang** [comment]
Updated docs:

Screenshot 2023-10-22 010310

**Review by ReenigneArcher — DISMISSED** [review]

File: src_assets/common/assets/web/config.html (hunk 3 above)
Comment (ReenigneArcher) [reply src_assets/common/assets/web/config.html]: Thanks, and I now see others that are out of order. I'll correct them in a separate PR.

File: src/config.h [hunk 4]
```diff
@@ -112,7 +112,8 @@ namespace config {
     std::chrono::duration<double> key_repeat_period;
 
     std::string gamepad;
-
+    bool ds4_back_as_touchpad_click;
+
```
Comment (ReenigneArcher) [inline src/config.h:116]:
```suggestion
    bool ds4_back_as_touchpad_click;

```

**Pull request merged**